src/main/java/org/bahmni/reports/template/TSIntegrationDiagnosisReportTemplate.java

```diff
@@ -0,0 +1,145 @@
+        if (report.getConfig().isDisplayTerminologyCode()) {
+            jasperReport.addColumn(col.column("Terminology Code", "Terminology Code", type.stringType()).setStyle(minimalColumnStyle).setHorizontalAlignment(HorizontalAlignment.CENTER));
+        }
+        if (report.getConfig().isDisplayGenderGroup()) {
+            jasperReport.addColumn(col.column("Female", "Female", type.stringType()).setStyle(minimalColumnStyle).setHorizontalAlignment(HorizontalAlignment.CENTER));
+            jasperReport.addColumn(col.column("Male", "Male", type.stringType()).setStyle(minimalColumnStyle).setHorizontalAlignment(HorizontalAlignment.CENTER));
+            jasperReport.addColumn(col.column("Other", "Other", type.stringType()).setStyle(minimalColumnStyle).setHorizontalAlignment(HorizontalAlignment.CENTER));
+            jasperReport.addColumn(col.column("Not disclosed", "Not disclosed", type.stringType()).setStyle(minimalColumnStyle).setHorizontalAlignment(HorizontalAlignment.CENTER));
+        }
+        jasperReport.addColumn(col.column("Total", "Total", type.stringType()).setStyle(minimalColumnStyle).setHorizontalAlignment(HorizontalAlignment.CENTER));
+        String formattedSql = getFormattedSql(sql, report.getConfig().getTsConceptSource(), startDate, endDate, tempTableName);
+        jasperReport.setShowColumnTitle(true).setWhenNoDataType(WhenNoDataType.ALL_SECTIONS_NO_DETAIL).setDataSource(formattedSql, connection);
+
+        return new BahmniReportBuilder(jasperReport);
+    }
+
+    private void loadTempTable(Connection connection, String tempTableName, String parentCode) {
+        int offset = 0;
+        int pageSize = getDefaultPageSize();
+        try {
+            String createSqlStmt = MessageFormat.format(CREATE_SQL_TEMPLATE, tempTableName);
+            String insertSqlStmt = MessageFormat.format(INSERT_SQL_TEMPLATE, tempTableName);
+            Statement statement = connection.createStatement();
+            statement.execute(createSqlStmt);
+            PreparedStatement pstmtInsert = connection.prepareStatement(insertSqlStmt);
+
+            TSPageObject pageObject = null;
+            do {
+                try {
+                    pageObject = fetchDescendantsByPagination(parentCode, pageSize, offset, "en");
+                } catch (IOException e) {
+                    throw new RuntimeException();
+                }
+                List<String> codes = pageObject.getCodes();
+                for (int batchcount = 0; batchcount < codes.size(); batchcount++) {
+                    pstmtInsert.setString(1, codes.get(batchcount));
+                    pstmtInsert.addBatch();
+                }
+                pstmtInsert.executeBatch();
+                offset += pageSize;
+            } while (offset < pageObject.getTotal());
+        } catch (SQLException e) {
+            logger.error("Error occured while making database call to " + tempTableName + " table");
+            throw new RuntimeException();
```

Comment:
What happens to temp tables if they were not deleted (due to RunTime Excepetion). They continue to stay in DB?

Reply:
Temp table is visible only within the current session, and is dropped automatically when the session is closed. More details are on this link - https://dev.mysql.com/doc/refman/5.7/en/create-temporary-table.html